toolkits/math/tests/test_arithmetic.py

```diff
@@ -2,49 +2,366 @@
+        factorial("1.0")
+    with pytest.raises(ToolExecutionError):
+        factorial("-1.0")
+    with pytest.raises(ToolExecutionError):
+        factorial("23.0")
+
+
+def test_deg_to_rad():
+    assert deg_to_rad("1") == "0.017453292519943295"
+    assert deg_to_rad("-1") == "-0.017453292519943295"
+    assert deg_to_rad("0") == "0.0"
+    assert deg_to_rad("-0") == "-0.0"
+    assert deg_to_rad("23") == "0.4014257279586958"
+    assert deg_to_rad("24") == "0.4188790204786391"
+    assert deg_to_rad("-10") == "-0.17453292519943295"
+    assert deg_to_rad("10") == "0.17453292519943295"
+    assert deg_to_rad("180") == "3.141592653589793"
+    assert deg_to_rad("0.0") == "0.0"
+    assert deg_to_rad("0.0000") == "0.0"
+    assert deg_to_rad("-0.0") == "-0.0"
+    assert deg_to_rad("1.0") == "0.017453292519943295"
+    assert deg_to_rad("-1.0") == "-0.017453292519943295"
+    assert deg_to_rad("23.0") == "0.4014257279586958"
+    assert deg_to_rad("0.4") == "0.006981317007977318"
+    assert deg_to_rad("-10.0") == "-0.17453292519943295"
+    assert deg_to_rad("10.0") == "0.17453292519943295"
+
+
+def test_rad_to_deg():
+    assert rad_to_deg("1") == "57.29577951308232"
+    assert rad_to_deg("-1") == "-57.29577951308232"
+    assert rad_to_deg("0") == "0.0"
+    assert rad_to_deg("-0") == "-0.0"
+    assert rad_to_deg("23") == "1317.8029288008934"
+    assert rad_to_deg("24") == "1375.0987083139757"
+    assert rad_to_deg("-10") == "-572.9577951308232"
+    assert rad_to_deg("10") == "572.9577951308232"
+    assert rad_to_deg("0.0") == "0.0"
+    assert rad_to_deg("0.0000") == "0.0"
+    assert rad_to_deg("-0.0") == "-0.0"
+    assert rad_to_deg("1.0") == "57.29577951308232"
+    assert rad_to_deg("-1.0") == "-57.29577951308232"
+    assert rad_to_deg("3.14") == "179.9087476710785"
+    assert rad_to_deg("0.4") == "22.918311805232932"
+    assert rad_to_deg("-10.0") == "-572.9577951308232"
+    assert rad_to_deg("10.0") == "572.9577951308232"
+
+
+def test_ceil():
+    assert ceil("1") == "1"
+    assert ceil("-1") == "-1"
+    assert ceil("0") == "0"
+    assert ceil("-0") == "0"
+    assert ceil("0.0") == "0"
+    assert ceil("0.0000") == "0"
+    assert ceil("-0.0") == "0"
+    assert ceil("1.0") == "1"
+    assert ceil("-1.0") == "-1"
+    assert ceil("3.14") == "4"
+    assert ceil("0.4") == "1"
+    assert ceil("-1.3") == "-1"
+
+
+def test_floor():
+    assert floor("1") == "1"
+    assert floor("-1") == "-1"
+    assert floor("0") == "0"
+    assert floor("-0") == "0"
+    assert floor("10") == "10"
+    assert floor("0.0") == "0"
+    assert floor("0.0000") == "0"
+    assert floor("-0.0") == "0"
+    assert floor("1.0") == "1"
+    assert floor("-1.0") == "-1"
+    assert floor("3.14") == "3"
+    assert floor("0.4") == "0"
+    assert floor("-1.3") == "-2"
+
+
+def test_round_num():
+    # TODO(mateo): ok with scientific notatin? ok with negative round digits?
+    assert round_num("1.2345", "-2") == "0E+2"
```

Comment:
In this case I think scientific notation looks silly, because it's such a small number. Is this a side effect of using `Decimal`?

Reply:
It is, it will return an integer if we use `float`. I will change it together with reorganizing the functions into math categories